test/TestCompoundMath.sol

```diff
@@ -1,93 +1,49 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.0;
 
+import {CompoundMathMock} from "./mocks/CompoundMathMock.sol";
+import {CompoundMathRef} from "./references/CompoundMathRef.sol";
 import "forge-std/Test.sol";
-import "src/math/CompoundMath.sol";
-import "./references/CompoundMathRef.sol";
-
-contract CompoundMathFunctions {
-    function mul(uint256 x, uint256 y) public pure returns (uint256) {
-        return CompoundMath.mul(x, y);
-    }
-
-    function div(uint256 x, uint256 y) public pure returns (uint256) {
-        return CompoundMath.div(x, y);
-    }
-}
-
-contract CompoundMathFunctionsRef {
-    function mul(uint256 x, uint256 y) public pure returns (uint256) {
-        return CompoundMathRef.mul(x, y);
-    }
-
-    function div(uint256 x, uint256 y) public pure returns (uint256) {
-        return CompoundMathRef.div(x, y);
-    }
-}
 
 contract TestCompoundMath is Test {
-    uint256 internal constant SCALE = 1e36;
-    uint256 internal constant WAD = 1e18;
-
-    CompoundMathFunctions compoundMath;
-    CompoundMathFunctionsRef compoundMathRef;
+    CompoundMathMock mock;
+    CompoundMathRef ref;
 
     function setUp() public {
-        compoundMath = new CompoundMathFunctions();
-        compoundMathRef = new CompoundMathFunctionsRef();
+        mock = new CompoundMathMock();
+        ref = new CompoundMathRef();
     }
 
     /// TESTS ///
 
     function testMul(uint256 x, uint256 y) public {
-        unchecked {
-            vm.assume(y == 0 || (x * y) / y == x);
-        }
+        vm.assume(y == 0 || x <= type(uint256).max / y);
 
-        assertEq(CompoundMath.mul(x, y), CompoundMathRef.mul(x, y));
+        assertEq(mock.mul(x, y), ref.mul(x, y));
     }
 
     function testMulOverflow(uint256 x, uint256 y) public {
-        unchecked {
-            vm.assume(y > 0 && (x * y) / y != x);
-        }
```

Comment:
This was also testing that the conditions were equivalent

Reply:
Indeed though I don't think we should include syntax tests inside our test suite?
I rather see this as a Solidity syntax test which should be included in a Solidity test suite